thirdparty/patches/boost-interlocked-prefer-intrinsics.patch

```diff
@@ -0,0 +1,10 @@
+diff --git boost/detail/interlocked.hpp boost/detail/interlocked.hpp
+--- boost/detail/interlocked.hpp
++++ boost/detail/interlocked.hpp
+@@ -21,1 +21,1 @@
+-#if defined( BOOST_USE_WINDOWS_H )
```

Comment:
I haven't looked at the context here, but we should avoid changing the behavior on non-windows (I assume that's what it is doing, since it replaces `BOOST_USE_WINDOWS_H` with `0`. Hard to tell what the repercussions are :)

Reply:
@pcmoritz Yeah, BOOST_USE_WINDOWS_H can only be true on Windows, and I'm setting it to false permanently. So it won't affect non-Windows.

Reply:
Got it yeah I was confused :)